### PR TITLE
NIFI-12351 Improve MiNiFi Service Scripts for Windows

### PR DESCRIPTION
**Pull request by rliszli**
# Summary

[NIFI-12351](https://issues.apache.org/jira/browse/NIFI-12351)

    This change contains the followings:
        - Apache Commons Daemon version update - Old version did not support non Oracle JDK installations. Now working with different JDK vendors and with latest Java versions as well.
        - fix missigng log file name properties, before this change there could be misleading log file names in the log directory
        - fixed install-service.bat, delete service.bat
        - Extended install service with 3 optinal parameters: serviceName, serviceUser, serviceUserPasswors
        - separated start-service and install-service
        - added stop-service.bat and stop-minifi.bat to make stopping easier 
        - updated status-minifi.bat, flowstatus-minifi.bat, and dump-minifi.bat scripts to startup properly

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [x] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [x] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [x] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [x] Pull Request based on current revision of the `main` branch
- [ ] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [x] Build completed using `mvn clean install -P contrib-check`
  - [x] JDK 21

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

**Comment by bejancsaba** [comment]
@rliszli Thanks for the change, without going deeper into the code aspect could you please add some details either to the ticket or the PR or even better to the README to explain what the change addresses, what is the intended behaviour change and how it can be validated?

**Comment by exceptionfactory** [comment]
Concurring with @bejancsaba, it would be helpful to elaborate on the goal of these changes. I updated the pull request title to reflect that fact that the changes related to service scripts for MiNiFi on Windows, but it would be helpful to summarize the intended improvements.

**Comment by rliszli** [comment]
Thanks for the suggestions and for the better naming. I added a short list for this PR containing the main elements of the change. I'll also update the `minifi-java-agent-quick-start.md` with a more detailed instructions on how to install as a service with the new scripts.

**Review by bejancsaba — COMMENTED** [review]
Thanks for the changes it is an area which clearly can benefit from some improvement. Maybe something is not right on my side but I'm getting the following
```
install-service.bat "serviceName=minifijava"
[2023-11-24 13:03:46] [error] [ 5116] The account name is invalid or does not exist, or the password is invalid for the account name specified.
```
The result is the same if I provide the serviceUser and servicePassword but I'm not an expert on win so maybe there is a simple answer here.

File: minifi/minifi-nar-bundles/minifi-framework-bundle/minifi-framework/minifi-resources/src/main/resources/bin/delete-service.bat [hunk 1]
```diff
@@ -16,8 +16,16 @@ rem    See the License for the specific language governing permissions and
...
 
-set SERVICE_NAME=minifi
-set SRV_BIN=%SERVICE_NAME%.exe
- 
-REM Remove service
-%SRV_BIN% //DS//%SERVICE_NAME%
+setlocal enabledelayedexpansion
+
+set arg1=%1
+if "!arg1:~1,11!" equ "serviceName" (
```
[the remark this answers is not in the document]
Comment (rliszli) [reply minifi/minifi-nar-bundles/minifi-framework-bundle/minifi-framework/minifi-resources/src/main/resources/bin/delete-service.bat:22]: Unfortunatelly I tried a lot of ways to do this. For example, the suggested _findstr_ cannot be applied inside an if condition. Honestly I don't like this either, as most part of how bat script working, but this was the "cleanest" or "simpliest" as I could find.

File: minifi/minifi-nar-bundles/minifi-framework-bundle/minifi-framework/minifi-resources/src/main/resources/bin/install-service.bat [hunk 2]
```diff
@@ -65,12 +125,21 @@ set LOG_PREFIX=minifi.log
 --StopMethod="%STOP_METHOD%" ^
 --StopTimeout="%STOP_TIMEOUT%" ^
 --Classpath="%CLASS_PATH%" ^
---JvmOptions="%JAVA_ARGS%" ^
+--JvmOptions9="%JAVA_ARGS%" ^
```
[the remark this answers is not in the document]
Comment (rliszli) [reply minifi/minifi-nar-bundles/minifi-framework-bundle/minifi-framework/minifi-resources/src/main/resources/bin/install-service.bat:116]: :)

**Comment by rliszli** [comment]
`Thanks for the changes it is an area which clearly can benefit from some improvement. Maybe something is not right on my side but I'm getting the following`
Previously I only tested it on Windows 10. After this included Windows Server 2019 as well. Now should work on both.

**Comment by exceptionfactory** [comment]
@rliszli This seems close to completion, aside from the one merge conflict. @bejancsaba I will defer to you for further review.

**Comment by exceptionfactory** [comment]
@rliszli and @bejancsaba any updates on bringing this pull request to completion?

**Comment by rliszli** [comment]
@exceptionfactory: We still have one issue that needs to be addressed. I'll close the PR until it gets resolved.

**Pull request closed without merging**